lib/reply.js

```diff
@@ -105,7 +105,7 @@ internals.Reply.prototype.interface = function (request, realm, options, next) {
         for (let i = 0; i < methods.length; ++i) {
             const method = methods[i];
             const decoration = this._decorations[method];
-            reply[method] = (domain ? domain.bind(decoration) : decoration);
+            reply[method] = domain && typeof decoration === 'function' ? domain.bind(decoration) : decoration;
```

Comment:
I would add () around `domain && typeof decoration === 'function'` to make it clearer
LGTM otherwise